src/libraries/Microsoft.Extensions.Configuration.Binder/tests/ConfigurationBinderTests.cs

```diff
@@ -1751,20 +1751,45 @@ public void CanBindNullableNestedStructProperties()
         }
 
         [Fact]
-        public void CanBindVirtualPropertiesWithoutDuplicates()
+        public void CanBindVirtualProperties()
         {
             ConfigurationBuilder configurationBuilder = new();
             configurationBuilder.AddInMemoryCollection(new Dictionary<string, string>
             {
-                { "Test:0", "1" }
+                { $"{nameof(BaseClassWithVirtualProperty.Test)}:0", "1" },
+                { $"{nameof(BaseClassWithVirtualProperty.TestGetSetOverriden)}", "2" },
+                { $"{nameof(BaseClassWithVirtualProperty.TestGetOverriden)}", "3" },
+                { $"{nameof(BaseClassWithVirtualProperty.TestSetOverriden)}", "4" },
+                { $"{nameof(BaseClassWithVirtualProperty.TestNoOverriden)}", "5" },
+                { $"{nameof(BaseClassWithVirtualProperty.TestVirtualSet)}", "6" }
             });
             IConfiguration config = configurationBuilder.Build();
 
             var test = new ClassOverridingVirtualProperty();
             config.Bind(test);
+
             Assert.Equal("1", Assert.Single(test.Test));
+            Assert.Equal("2", test.TestGetSetOverriden);
+            Assert.Equal("3", test.TestGetOverriden);
+            Assert.Equal("4", test.TestSetOverriden);
+            Assert.Equal("5", test.TestNoOverriden);
+            Assert.Null(test.ExposeTestVirtualSet());
```

Comment:
Why is this Null instead of `6`?

Reply:
Because set-only property was not bound, which was existing behavior (it is not about code in `GetAllProperties()`). I added this test case to make sure all get/set combinations covered and behavior not break. @eerhardt